### DISCUSSION TITLE
Allow setting default parallel amount for each task

### DISCUSSION THREAD

**Discussion by koliveira15** [nrwl/nx#23416] [Feature Requests]
- [ ] I'd be willing to implement this feature ([contributing guide](https://github.com/nrwl/nx/blob/master/CONTRIBUTING.md))

## Description
Add the ability to specify the number of default parallel projects per build task

## Motivation
Some tasks require different default number of parallel executions. My test task might differ from my build task based on CI constraints.

## Suggested Implementation
Leverage the new nx.json schema where each task has its own configuration. The default parallel property could be added here.

```json
// nx.json
...
"targetDefaults": {
    "build": {
      "dependsOn": ["^build"],
      "inputs": ["production", "^production"],
      "cache": true,
      "parallel": 3
    },
    "test": {
      "inputs": ["default", "^production", "{workspaceRoot}/jest.preset.js"],
      "dependsOn": ["^test"],
      "cache": true,
      "parallel": 5
    }
  },
...
```

**Comment by mattfysh** [comment]
To add another use case, I use the `@wanews/nx-pulumi` plugin in a number of projects and it requires stdin be inherited by the child process for interactive prompts. In CI environments it can run non-interactively, but when I'm developing I do want to interact with the task as it runs

This implementation would allow me to specify `"parallel": 1` perhaps in nx.json, and ensure there is no stdin confusion

**Comment by FrozenPandaz** [comment]
Making this discussion to discuss more how people would like to configure this.

**Comment by unional** [comment]
I want to do the same thing. The OP proposal look reasonable.
